fidesctl/src/fidesctl/core/generate_dataset.py

```diff
@@ -116,35 +134,143 @@ def create_dataset_collections(
     return table_manifests
 
 
-def create_dataset(engine: Engine, collections: List[DatasetCollection]) -> Dataset:
+def find_uncategorized_dataset_fields(
+    dataset_key: str, dataset: Optional[Dataset], db_dataset: Dict[str, List[str]]
+) -> Tuple[List[str], int]:
+    """
+    Given an object that represents a database dataset definition, finds
+    uncategorized keys and coverage ratio.
+    """
+    uncategorized_fields = []
+    total_field_count = 0
+
+    for db_dataset_collection in db_dataset.keys():
+        dataset_collection = (
+            next(
+                (
+                    colection
+                    for colection in dataset.collections
+                    if colection.name == db_dataset_collection
+                ),
+                None,
+            )
+            if dataset
+            else None
+        )
+
+        for db_dataset_field in db_dataset.get(db_dataset_collection, []):
+            total_field_count += 1
+            field_found = (
+                any(
+                    field.name == db_dataset_field and field.data_categories
+                    for field in dataset_collection.fields
+                )
+                if dataset_collection
+                else False
+            )
+
+            if not field_found:
+                uncategorized_fields.append(
+                    f"{dataset_key}.{db_dataset_collection}.{db_dataset_field}"
+                )
```

Comment:
Ah, `field_found` was a refactoring miss because we were originally just checking for that. Thanks.

I thought about your suggestion and was on the fence at first but i do think it's most logical to be searching for `field_uncategorized` as opposed to `field_categorized`.  Only thing missing is the change to `else True`

Reply:
woops, should also be `or` so... `field.name != db_dataset_field or not field.data_categories`